packages/langchain/src/openai/util.test.ts

```diff
@@ -2,55 +2,71 @@ import {
   AzureOpenAiCreateChatCompletionResponse,
   AzureOpenAiCreateChatCompletionRequest
 } from '@sap-ai-sdk/foundation-models';
-import nock from 'nock';
-import { BaseMessage, HumanMessage } from '@langchain/core/messages';
 import {
-  mockClientCredentialsGrantCall,
-  parseMockResponse
-} from '../../../../test-util/mock-http.js';
+  BaseMessage,
+  RemoveMessage,
+  ToolMessage
+} from '@langchain/core/messages';
+import { parseMockResponse } from '../../../../test-util/mock-http.js';
 import { mapLangchainToAiClient, mapOutputToChatResult } from './util.js';
 import { AzureOpenAiChatClient } from './chat.js';
+import { AzureOpenAiChatCallOptions } from './types.js';
 
 const openAiMockResponse =
   parseMockResponse<AzureOpenAiCreateChatCompletionResponse>(
     'foundation-models',
     'azure-openai-chat-completion-success-response.json'
   );
 
-describe('Mapping Functions', () => {
-  beforeEach(() => {
-    mockClientCredentialsGrantCall();
-  });
-
-  afterEach(() => {
-    nock.cleanAll();
-  });
+// Signal and Prompt Index are provided by the super class in every call
+const defaultOptions = {
+  signal: undefined,
+  promptIndex: 0
+};
 
+describe('Mapping Functions', () => {
   it('should parse an OpenAI response to a (LangChain) chat response', async () => {
     const result = mapOutputToChatResult(openAiMockResponse);
     expect(result).toMatchSnapshot();
   });
 
   it('should parse a LangChain input to an AI SDK input', async () => {
     const langchainPrompt: BaseMessage[] = [
-      new HumanMessage('Where is the deepest place on earth located')
+      new ToolMessage('Test Content', 'test-id')
```

Comment:
how about leaving the human message in and rather adding other cases?

Reply:
I felt like it was a redundancy, as it traverses the same functions but less than the tool message, what do you think?